executor/mock/jobrunner.go

```diff
@@ -181,14 +188,40 @@ func NewJobRunner() *JobRunner {
 	}
 
 	jobRunner := &JobRunner{
-		ctx: ctx, cancel: cancel,
+		ctx:             ctx,
+		cancel:          cancel,
 		runner:          r,
 		shutdownChannel: make(chan struct{}),
 	}
 
 	return jobRunner
 }
 
+// DockerPull does a docker pull via DockerRuntime
+func DockerPull(imgName string, version string) (bool, error) {
+	cfg, dockerCfg := generateConfigs()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	rt, err := docker.NewDockerRuntime(ctx, metrics.Discard, *dockerCfg, *cfg)
```

Comment:
Add `defer cancel()`, that way cancel() gets called any time the stack unwinds. It's generally a good pattern.